pkg/controller/machinepool/gcpactuator.go

```diff
@@ -471,38 +469,20 @@ func getNetwork(remoteMachineSets []machineapi.MachineSet,
 		return "", "", nil
 	}
 
-	remoteMachineProviderSpec, err := decodeGCPMachineProviderSpec(
+	remoteMachineProviderSpec, err := gcpprovider.ProviderSpecFromRawExtension(
 		remoteMachineSets[0].Spec.Template.Spec.ProviderSpec.Value,
-		scheme,
 	)
 	if err != nil {
 		logger.WithError(err).Warn("cannot decode GCPMachineProviderSpec from remote machinesets")
 		return "", "", errors.Wrap(err, "cannot decode GCPMachineProviderSpec from remote machinesets")
 	}
 
 	if len(remoteMachineProviderSpec.NetworkInterfaces) == 0 {
-		logger.Warn("remote machine do not have any network interfaces")
-		return "", "", errors.Wrap(err, "remote machine do not have any network interfaces")
+		logger.Error("remote machine does not have any network interfaces")
+		return "", "", errors.New("remote machine does not have any network interfaces")
```

Comment:
I've changed this to return an error. Previously the errors.Wrap would always be nil if we were able to decode the GCPMachineProviderSpec successfully but there were no network interfaces.

Reply:
Should that be a separate PR?

Reply:
The reason this needs to return an error is because `ProviderSpecFromRawExtension(rawExtension)` will give you back an empty object if your `rawExension` was `nil` which we do in our test here https://github.com/openshift/hive/blob/master/pkg/controller/machinepool/gcpactuator_test.go#L700.

Reply:
Got it.